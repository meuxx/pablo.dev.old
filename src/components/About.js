import React from 'react'

class About extends React.Component {
  render() {
    return (
      <section id="one">
        <h2>About me</h2>
        <p>
          I'm <strong>Pablo Maceda</strong>, back-end developer offering 5 years
          of experience, with a solid background as PHP developer and a broad
          technical skill set. I have a passion for good design, clean code and
          good performance.
        </p>
        <p>
          As a self-taught developer, I always enjoy learning new tools and
          technologies.
        </p>

        <ul className="actions">
          <li>
            <a
              href="https://cv.pablo.dev"
              className="button"
              rel="noopener noreferrer"
              target="_blank"
            >
              View CV
            </a>
          </li>
        </ul>
      </section>
    )
  }
}

export default About
