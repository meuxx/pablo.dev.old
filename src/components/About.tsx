import React from 'react'
import { useGoal } from 'gatsby-plugin-fathom'
import { Section, SectionHeader, Button } from '../styles/styles'

const About: React.FC = () => {
  const handleGoal = useGoal('V5LKNTOU')

  return (
    <Section>
      <SectionHeader>
        <h2>About me</h2>
      </SectionHeader>
      <p>
        I&apos;m <strong>Pablo Maceda</strong>, back-end developer offering 5+ years of experience, with a solid
        background as PHP developer currently working with NodeJS. I have a broad technical skill set, passion for clean
        code and good performance.
      </p>
      <p>As a self-taught developer, I always enjoy learning new tools and technologies.</p>

      <Button
        href="https://cv.pablo.dev"
        className="button"
        rel="noopener noreferrer"
        target="_blank"
        aria-label="Curriculum Vitae"
        title="Curriculum Vitae"
        onClick={handleGoal}
      >
        View CV
      </Button>
    </Section>
  )
}

export default About
