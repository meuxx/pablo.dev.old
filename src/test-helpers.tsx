import { render } from '@testing-library/react'
import { Themed } from './styles/GlobalStyle'
import React from 'react'

const renderComponent: typeof render = (ui, options) => render(ui, { wrapper: Themed, ...options })

const mockComponent = (componentName: string): React.FC => (props) => {
  const { children, ...otherProps } = props
  return (
    <div data-component={componentName} data-props={JSON.stringify(otherProps, null, ' ')}>
      {children}
    </div>
  )
}

export { renderComponent, mockComponent }
