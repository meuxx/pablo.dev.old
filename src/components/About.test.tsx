import React from 'react'
import { renderComponent } from '../test-helpers'
import About from './About'

describe('About', () => {
  it('renders without crashing', () => {
    const { container } = renderComponent(<About />)

    expect(container).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})
