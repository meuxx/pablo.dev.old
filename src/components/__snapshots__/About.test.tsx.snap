// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders without crashing 1`] = `
.c0 {
  margin: 2em 0 0 0;
  padding: 2em 0 0 0;
}

.c0:first-child {
  margin-top: 0;
  padding-top: 0;
}

.c0 p {
  color: #b2b2b2;
  position: relative;
  margin: 0 0 2em 0;
}

.c0 p strong {
  color: #ffffff;
  font-weight: 700;
}

.c1 h1,
.c1 h2 {
  color: #ffffff;
  font-weight: 700;
}

.c1 h1 {
  margin: 0 0 1em;
  line-height: 1.5em;
}

.c1 h2 {
  margin: 0 0 1em;
  font-size: 1.5em;
  line-height: 1.5em;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out, border-color 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,color 0.2s ease-in-out, border-color 0.2s ease-in-out;
  background-color: transparent;
  border-radius: 0.35em;
  border: solid 3px #0B61A4;
  color: #0B61A4;
  cursor: pointer;
  display: inline-block;
  font-weight: 700;
  height: calc(2.75em + 6px);
  line-height: 2.75em;
  min-width: 10em;
  padding: 0 1.5em;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
}

.c2:hover {
  background-color: rgba(3,62,107,0.2);
}

.c2:active {
  background-color: rgba(3,62,107,0.3);
}

@media screen and (max-width:736px) {
  .c0 {
    margin: 1.5em 0 0 0;
    padding: 1.5em 0 0 0;
  }
}

@media screen and (max-width:736px) {
  .c1 h2 {
    font-size: 1.2em;
  }
}

<div>
  <section
    class="c0"
  >
    <header
      class="c1"
    >
      <h2>
        About me
      </h2>
    </header>
    <p>
      I'm 
      <strong>
        Pablo Maceda
      </strong>
      , back-end developer offering 5+ years of experience, with a solid background as PHP developer currently working with NodeJS. I have a broad technical skill set, passion for clean code and good performance.
    </p>
    <p>
      As a self-taught developer, I always enjoy learning new tools and technologies.
    </p>
    <a
      aria-label="Curriculum Vitae"
      class="c2 button"
      href="https://cv.pablo.dev"
      rel="noopener noreferrer"
      target="_blank"
      title="Curriculum Vitae"
    >
      View CV
    </a>
  </section>
</div>
`;
